### PR TITLE
Closes #10

### LINKED ISSUE #10: Handler system
Need to create some struct / interfaces for handling the handlers. Probably use some form of HOF as per @periaate suggestion.

### PR DESCRIPTION
**Pull request by Xeckt**
New implementations:
- Handler system for session events
- Created internal folder and moved most systems there
- Added intent handlers system for intent specification
- Fixed cleanup func with waitgroup syncs

Thanks for showing me how dependency injection works here @periaate

### CODE REVIEW

File: manager.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
```
Comment (periaate) [inline manager.go:3]: Imports haven't been formatted by gopls.

File: member.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package main
+
+import (
```
Comment (periaate) [inline member.go:3]: Imports haven't been formatted by gopls.

File: message.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package main
+
+import (
```
Comment (periaate) [inline message.go:3]: Imports haven't been formatted by gopls.

File: intents.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	case discordgo.IntentGuildScheduledEvents:
+		return "GuildScheduledEvents"
+	case discordgo.IntentAutoModerationConfiguration:
+		return "AutoModerationConfiguration"
+	case discordgo.IntentAutoModerationExecution:
+		return "AutoModerationExecution"
+	default:
+		return "No string found for this intent"
+	}
+	return "None found"
```
Comment (periaate) [inline intents.go:48]: This return statement is unreachable.

**Review by periaate — CHANGES_REQUESTED** [review]
There are a few formatting problems and an unreachable branch. Otherwise, looks good to me! 👍

File: intents.go (hunk 4 above)
Comment (Xeckt) [reply intents.go]: Resolved.

**Review by periaate — APPROVED** [review]

**Pull request merged**